core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -201,6 +252,93 @@ static ExprValue getFormattedTime(ExprValue timeExpr, ExprValue formatExpr) {
     return getFormattedString(formatExpr, TIME_HANDLERS, time);
   }
 
+  private static boolean canGetDate(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.YEAR)
+        || !ta.isSupported(ChronoField.MONTH_OF_YEAR)
+        || !ta.isSupported(ChronoField.DAY_OF_MONTH)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean canGetTime(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.HOUR_OF_DAY)
+        || !ta.isSupported(ChronoField.MINUTE_OF_HOUR)
+        || !ta.isSupported(ChronoField.SECOND_OF_MINUTE)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean isValidTemporalAccessor(TemporalAccessor ta) {
+    if (canGetDate(ta) || canGetTime(ta)) {
+      return true;
+    }
+    return false;
+  }
+
+  static ExprValue parseStringWithDateOrTime(FunctionProperties fp,
+                                             ExprValue datetimeStringExpr,
+                                             ExprValue formatExpr) {
+
+    //Replace patterns with % for Java DateTimeFormatter
+    StringBuffer cleanFormat = getCleanFormat(formatExpr);
+    final Matcher matcher = pattern.matcher(cleanFormat.toString());
+    final StringBuffer format = new StringBuffer();
+
+    while (matcher.find()) {
+      matcher.appendReplacement(format,
+          STR_TO_DATE_FORMATS.getOrDefault(matcher.group(),
+              String.format("'%s'", matcher.group().replaceFirst(MOD_LITERAL, ""))));
+    }
+    matcher.appendTail(format);
+
+    TemporalAccessor taWithMissingFields;
+    //Return NULL for invalid parse in string to align with MySQL
+    try {
+      //Get Temporal Accessor to initially parse string without default values
+      taWithMissingFields = new DateTimeFormatterBuilder()
+          .appendPattern(format.toString()).toFormatter().parse(datetimeStringExpr.stringValue());
+
+      if (!isValidTemporalAccessor(taWithMissingFields)) {
+        throw new DateTimeException("Not enough data to build a valid Date, Time, or Datetime.");
+      }
+    } catch (DateTimeException e) {
+      return ExprNullValue.of();
```

Comment:
Maybe: return `NULL` in both cases or throw in both cases. MySQL returns NULL, but I prefer exception - it is move verbose.
TBD.

Reply:
Implementation will be kept as it is after discussing it this morning.